### PR TITLE
scheduler: Instead of directly assigning s.isLeader, use HandleLeaderChange to populate logs correctly

### PR DESCRIPTION
**Pull request by jzila**
Signed-off-by: John Zila <john@jzila.com>

### CODE REVIEW

File: controller/scheduler/scheduler.go [hunk 1]
```diff
@@ -257,10 +257,12 @@ func (s *Scheduler) Run() error {
 	}
 
 	var err error
-	s.isLeader, err = s.discoverd.Register()
+	var isLeader bool
+	isLeader, err = s.discoverd.Register()
```
Comment (lmars) [inline controller/scheduler/scheduler.go]: Just use `:=`

Comment (jzila) [inline controller/scheduler/scheduler.go]: Already done

File: controller/scheduler/scheduler.go [hunk 2]
```diff
@@ -256,11 +256,11 @@ func (s *Scheduler) Run() error {
 		return err
 	}
 
-	var err error
-	s.isLeader, err = s.discoverd.Register()
+	isLeader, err := s.discoverd.Register()
 	if err != nil {
 		return err
 	}
+	s.HandleLeaderChange(isLeader)
```
Comment (lmars) [inline controller/scheduler/scheduler.go]: This has the side effect of calling [rectifyAll](https://github.com/jzila/flynn/blob/scheduler-v2/controller/scheduler/scheduler.go#L521), should we not just leave it as it was and log a message here?

Comment (jzila) [inline controller/scheduler/scheduler.go]: Nope, we want `rectifyAll` to be called if it's the new leader.

Comment (jzila) [inline controller/scheduler/scheduler.go]: Oh I see the formations will won't be populated yet. Hmm.

Comment (jzila) [inline controller/scheduler/scheduler.go]: OK so right now the `rectifyAll` call will be a harmless no-op. I guess we'll let the formation stream worry about what to rectify.

**Comment by lmars** [comment]
LGTM

**Pull request merged**